### ISSUE TITLE
Sidebar navigation: Jetpack sites missing several menu items

### ISSUE THREAD

**Issue by creativecoder** [Automattic/wp-calypso#66068]
### Quick summary

Accessing a Jetpack (non-Atomic) site in Calypso, there seem to be several discrepancies in the sidebar navigation menu:

Some of the sub-menus do not show when hovering (desktop) or tapping (mobile). I suspect these are menus that we don't inject any submenu items into from Calypso.

- Posts
- Pages
- Comments
- Feedback
- Plugins
- Users

Some menu items are missing (they show in wp-admin, but not in Calypso)

- Appearance 
    - Editor (beta)
    - Additional CSS
- Tools
    - Available Tools
    - Site Health
    - Export Personal Data
    - Erase Personal Data
    - Theme File Editor
    - Plugin File Editor
- Settings
    - Reading
    - Media
    - Permalinks 
    - Privacy

### Steps to reproduce

- Create and connect a new Jetpack site (jurassic.ninja)
- Load the site in Calypso

### What you expected to happen

To see all of the available menu options in the sidebar navigation 

### What actually happened

Missing menu items

### Browser

Mozilla Firefox

### Platform (Simple, Atomic, or both?)

Self-hosted

### Other notes

I know Jetpack (non-Atomic) sites did not have Nav unification turned on when it launched.

It's almost as if Nav unification was accidentally turned on for these sites without the proper feature support.

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

Yes, easy to implement

### Workaround details

Can visit wp-admin, instead, but it's a very poor experience in Calypso.

**Comment by jeherve** [comment]
I assume this is related to the fallback menu we added to Jetpack sites when we worked on nav-unification: #49352

@mmtr Do you know if someone has picked up maintenance of nav-unification since Ajax moved on to different projects? What would be the expected behaviour here?

**Comment by jeherve** [comment]
Somewhat related issues:

- #62975 
- #64935

**Comment by mmtr** [comment]
I think the behavior noted in the summary is the expected behavior.

The Nav Unification project was a combination of multiple things:

1.  New sidebar UI in Calypso to make it visually similar to the WP Admin menu
2. Fetch menu items from an API endpoint (so Calypso no longer registers them)
3. Make menu API endpoint to return the same menu items that are registered in WP Admin

When we launched Nav Unification, we received feedback against enabling it on Jetpack sites (p1HpG7-bjx-p2) because we want to have a clear distinction between Calypso and WP Admin on those sites.

However, that increased the maintenance cost (2 different sidebars to maintain in Calypso), so we decided to only activate the items 1 and 2 from the list above on Jetpack sites. The menu API endpoint for Jetpack sites still return the same menu items that were registered in the old Calypso sidebar and deliberately exclude menu items that link to WP Admin. See p1HpG7-c19-p2.

**Comment by mmtr** [comment]
> Do you know if someone has picked up maintenance of nav-unification since Ajax moved on to different projects?

I'm not aware of anyone, so I think this would fall into the @Automattic/dotcom-janitors scope

**Comment by mpkelly** [comment]
I am going to look at this one now.

**Comment by mmtr** [comment]
@mpkelly make sure to coordinate with the Jetpack teams before adding any menu item to the Calypso sidebar that points to WP Admin, since that's something that conflicts with the Calypso blue/green strategy for Jetpack sites (aka `wordpress.com` vs `cloud.jetpack.com`, see p1HpG7-bcl-p2#comment-44627 and p6TEKc-4ar-p2).

IMO, I think we should we close this issue as "won't do" for now, since the missing menu items noted in the issue summary is an intentional behavior.

**Comment by mpkelly** [comment]
Thanks, @mmtr. @daledupreez already told me the same in our Slack Channel - we need to discuss and agree on any changes first. 

If this is expected behaviour then maybe @creativecoder would consider closing it down.

**Comment by creativecoder** [comment]
> IMO, I think we should we close this issue as "won't do" for now, since the missing menu items noted in the issue summary is an intentional behavior.

I strongly disagree. While most Jetpack users don't use Calypso and go directly to wp-admin, I don't think that's a reason to leave the Calypso interface for Jetpack sites in a broken state. If we're going to deprecate using Calypso for Jetpack (non-Atomic) sites, let's do that and remove it, not leave it in a broken state for users to stumble upon.
